### ISSUE TITLE
Is it possible to know that the VGSTextField (with card number type) fully filled in and not valid?

### ISSUE THREAD

**Issue by thinhv** [verygoodsecurity/vgs-collect-ios#121]
Hi, In our app we need to show a popover view when the card number is fully filled and not valid. However VGSTextField does not provide that value. We can't just rely on `isValid` since `isValid` is `false` when textField is empty and we don't want to show the popover if it is empty.

Thank you!

**Comment by dmytrokhl** [comment]
Hi @thinhv It's possible for you to check if VGSTextField is empty.

VGSTextfield’s State has **isEmpty** attribute https://verygoodsecurity.github.io/vgs-collect-ios/Classes/State.html#/s:13VGSCollectSDK5StateC7isEmptySbvp

You can combine it with **isValid** attribute and do the checks like:

```
if !textfield.state.isValid && textfield.state.isEmpty {
    dontShowPopup()
}
```

Will it works for you?

**Comment by thinhv** [comment]
Hi @dmytrokhl thanks for the suggestion, but our designers don't want to show the popup while the textField is being filled. Only when text field is fully filled. Something like this:
```

**Comment by dmytrokhl** [comment]
Will that work for you if we add raw input length attribute and make `possibleLength()` public? So you can get from textfield.state `cardBrand` and `user input length`,  and then compare it with `possibleLength()`.

Regarding `getting text from the private objects` - if you need to be in `PCI` scope you can't touch/save/edit sensitive data by yourself or edit the SDK.

**Comment by thinhv** [comment]
Thanks. raw input length and `possibleLength() ` would solve our problem.
Sorry to bring it here but of course I don't edit the SDK code. I can easily get `text` property on `MaskedTextField`'s super class. 
Anyway, I don't want to touch it that why I created this issue.

**Comment by dmytrokhl** [comment]
@thinhv  this has been released in `VGSCollectSDK` [v1.5.0](https://github.com/verygoodsecurity/vgs-collect-ios/releases/tag/1.5.0). You can check `textfield.state.cardBrand.cardLengths` &  `textfield.state.inputLength` fields.
